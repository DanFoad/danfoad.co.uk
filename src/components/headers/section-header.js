import React from 'react';
import PropTypes from 'prop-types';

export class SectionHeader extends React.Component {
  render() {
    return (
      <div className="section-header">
        <h2>{this.props.children}</h2>
      </div>
    );
  }
}

SectionHeader.propTypes = {
  fontSize: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
    .isRequired,
};
