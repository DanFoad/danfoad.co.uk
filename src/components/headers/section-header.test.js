import React from 'react';
import { shallow } from 'enzyme';

import { SectionHeader } from './section-header';

describe('section-header', () => {
  let props;

  beforeEach(() => {
    props = {
      fontSize: 40,
      children: 'Test',
    };
  });

  it('should match snapshot for string as child', () => {
    expect(shallow(<SectionHeader {...props} />)).toMatchSnapshot();
  });

  it('should match snapshot for element as child', () => {
    expect(
      shallow(
        <SectionHeader {...props}>
          <span>Test</span>
        </SectionHeader>
      )
    ).toMatchSnapshot();
  });
});
