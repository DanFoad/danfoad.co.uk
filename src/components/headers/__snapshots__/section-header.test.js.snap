// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section-header should match snapshot for element as child 1`] = `
<div
  className="section-header"
>
  <h2>
    <span>
      Test
    </span>
  </h2>
</div>
`;

exports[`section-header should match snapshot for string as child 1`] = `
<div
  className="section-header"
>
  <h2>
    Test
  </h2>
</div>
`;
